src/platform/graphics/linux_dmabuf.cpp

```diff
@@ -856,6 +858,78 @@ class LinuxDmaBufParams : public mir::wayland::LinuxBufferParamsV1
     }
 };
 
+class LinuxDmaBufFeedback : public mir::wayland::LinuxDmabufFeedbackV1
+{
+public:
+    LinuxDmaBufFeedback(
+        wl_resource* new_resource,
+        struct wl_resource* surface,
+        std::shared_ptr<mg::DMABufEGLProvider> provider)
+        : mir::wayland::LinuxDmabufFeedbackV1(new_resource, Version<5>{}),
+          provider{std::move(provider)}
+    {
+        // We have the same feedback regardless of surface.
+        (void)surface;
+
+        // Build table in shared memory.
+        struct Format
+        {
+            uint32_t format;
+            uint32_t padding; /* unused */
+            uint64_t modifier;
+        };
+        auto const& formats = this->provider->supported_formats();
+        size_t format_table_length = 0;
+        for (auto i = 0u; i < formats.num_formats(); ++i)
+        {
+            format_table_length += formats[i].modifiers.size();
+        }
+        mir::AnonymousShmFile shm_buffer{format_table_length * sizeof(Format)};
+        Format *data = static_cast<Format*>(shm_buffer.base_ptr());
+        size_t k = 0;
+        for (auto i = 0u; i < formats.num_formats(); ++i)
+        {
+            auto format = formats[i];
+            for (auto j = 0u; j < format.modifiers.size(); ++j)
+            {
+                data[k].format = format.format;
+                data[k].padding = 0;
+                data[k].modifier = format.modifiers[j];
+                k++;
+            }
+        }
```

Comment:
Hm. I was *going* to suggest some fancier range-based C++ here, but that's not possible because `DmaBufFormatDescriptors` doesn't model a range.

And now is not the time to be implementing that.